tsl/test/expected/bgw_policy.out

```diff
@@ -437,10 +439,256 @@ WARNING:  unexpected interval: smaller than one second
+ test_drop_chunks 
+------------------
+ 
+(1 row)
+
+-- added one to now() so time entry with value 0 should be dropped now
+SELECT count(*) FROM _timescaledb_catalog.chunk as c, _timescaledb_catalog.hypertable as ht where c.hypertable_id = ht.id and ht.table_name='test_table_int' \gset after_
+select :before_count=:after_count+1;
+ ?column? 
+----------
+ t
+(1 row)
+
+select * from test_table_int;
+ time | junk 
+------+------
+    1 |    1
+    2 |    2
+    3 |    3
+   42 |   42
+(4 rows)
+
+-- make the now() function invalid -- returns INT and not BIGINT
+drop function dummy_now2();
+create or replace function dummy_now2() returns INT LANGUAGE SQL IMMUTABLE as  'SELECT 2::INT';
+\set ON_ERROR_STOP 0
+select test_drop_chunks(:drop_chunks_job_id);
+ERROR:  invalid integer_now function
+\set ON_ERROR_STOP 1
+-- todo:: See info about the failed job, it does not appear here ` select * from  _timescaledb_internal.bgw_job_stat;
+-- test the case when partitioning function and now function are set.
+create table part_time_now_func(time float8, temp float8);
+create or replace function time_partfunc(unixtime float8)
+    returns bigint language plpgsql immutable as
+$body$
+declare
+    retval bigint;
+begin
+
+    retval := unixtime::bigint;
+    raise notice 'time value for % is %', unixtime, retval;
+    return retval;
+end
+$body$;
+create or replace function dummy_now() returns bigint language sql immutable as  'select 2::bigint';
+select create_hypertable('part_time_now_func', 'time', time_partitioning_func => 'time_partfunc', chunk_time_interval=>1);
+NOTICE:  adding not-null constraint to column "time"
+        create_hypertable        
+---------------------------------
+ (4,public,part_time_now_func,t)
+(1 row)
+
+insert into part_time_now_func values
+(1.1, 23.4), (2.2, 22.3), (3.3, 42.3);
+NOTICE:  time value for 1.1 is 1
+NOTICE:  time value for 1.1 is 1
+NOTICE:  time value for 1.1 is 1
+NOTICE:  time value for 1.1 is 1
+NOTICE:  time value for 2.2 is 2
+NOTICE:  time value for 2.2 is 2
+NOTICE:  time value for 2.2 is 2
+NOTICE:  time value for 2.2 is 2
+NOTICE:  time value for 3.3 is 3
+NOTICE:  time value for 3.3 is 3
+NOTICE:  time value for 3.3 is 3
+NOTICE:  time value for 3.3 is 3
+select * from part_time_now_func;
+ time | temp 
+------+------
+  1.1 | 23.4
+  2.2 | 22.3
+  3.3 | 42.3
+(3 rows)
+
+select set_integer_now_func('part_time_now_func', 'dummy_now');
+ set_integer_now_func 
+----------------------
+ 
+(1 row)
+
+select add_drop_chunks_policy('part_time_now_func', 0) as drop_chunks_job_id \gset
+select test_drop_chunks(:drop_chunks_job_id);
+ test_drop_chunks 
+------------------
+ 
+(1 row)
+
+select * from part_time_now_func;
+ time | temp 
+------+------
+  2.2 | 22.3
+  3.3 | 42.3
+(2 rows)
+
+select remove_drop_chunks_policy('part_time_now_func');
+ remove_drop_chunks_policy 
+---------------------------
+ 
+(1 row)
+
+\c :TEST_DBNAME :ROLE_SUPERUSER
+alter schema public rename to new_public;
```

Comment:
can we also test ALTER FUNCTION...RENAME TO and make sure the catalogs are update when you rename the function?

Reply:
hm, renaming function does not change the catalogs and of course later attempts to use the function fail. This fails for partitioning function as well.

Reply:
DISCUSS

Reply:
Maybe better left to a separate PR then